### PR TITLE
UI character implementation

### LINKED ISSUE #1397: [FEATURE] Fix character rotation speed in menu
## 🚀 Feature

- it's too fast
- also increase the box in which you are able to rotate the character

### LINKED ISSUE #1475: [FEATURE] Fix player location
## 🚀 Feature

The current issue is a discrepancy in the character's location between the main screen and the character info screen. 
One possible solution is to try using the same location for both screens.

### References
Yellow line: main screen
Red line: character info screen 
muflus
h4ck
uma

### LINKED ISSUE #1604: [BUG] Character rotation in main screen link fix
## 🐛 Bug Report

Character rotation in main screen sometimes triggers a goes to character detail

### Steps to Reproduce

Press the character, rotate it and release the finger in a similar position the press down started

### LINKED ISSUE #1605: [FEATURE] Make character models in main and character info screens have the same ux features
### Motivation

Main screen and Character info screens have 2 different behaviors when it comes to rotating the character

### Implementation

The character rotation for UI models should all be the same

### PR DESCRIPTION
**Pull request by AngieDutra**
Closes #1397
Closes #1605
Closes #1604 
Closes #1475 

## Motivation

Bugs in main:
- Rotation speed was too high
- Two ways of rendering the models in the UI (one in main and the other in character info) where used instead of one prefab. This resulted in two UX. mainly noticeable in the rotation feature. In main the rotation was within the character and in character info the rotation could be trigger from the sides
- When the user pressed the character, then dragged the finger but for the release of it they returned to the original position the link would trigger. This is not ideal behavior.

## Summary of changes
Solved issues:
- Rotation speed was adjusted
- Now we have 1 prefab used to render UI models
- The code was refactored to have a clean and readable code

In addition:
- Adjusted the main screen and characters info model position to be the same in both screens and don't have weird offsets from screen to screen

## How has this been tested?

- Use the rotation feature in character info and in main screen
- Use the link feature in main screen to go to character info
- Try to use it as random as possible to find any issues I could've missed

## Checklist
- [x] I have tested the changes locally.
- [x] I self-reviewed the changes on GitHub, line by line.
- [ ] Tests have been added/updated.
- [ ] This change requires new documentation.
  - [ ] Documentation has been added/updated.
- [x] I have tested the changes in another devices.
  - [x] Tested in iOS.
  - [ ] Tested in Android.

### CODE REVIEW

**Review by BertovDev — CHANGES_REQUESTED** [review]
Not working as expected. 
https://github.com/lambdaclass/curse_of_mirra/assets/62818001/5d5b046b-1c02-4396-896d-5649b9aedeab

**Review by BertovDev — COMMENTED** [review]
This is awesome and just a few comments.
In my opinion the height of the rotate container is too much since it feels weird to touch the part of the screen marked in the photo and rotate the character.

Screenshot 2024-03-19 at 19 36 15

File: client/Assets/Scripts/UI/ModelLink.cs [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+using UnityEngine;
+using UnityEngine.EventSystems;
+
+public class ModelLink : MMTouchButton, IDragHandler
+{
+    [SerializeField]
+    RotateUIModel uiModelRotator;
+    bool isInsideContainer;
+    Vector3 touchStartPos;
+    private const float MIN_DIFFERENCE = 10.0f;
```
Comment (BertovDev) [inline client/Assets/Scripts/UI/ModelLink.cs:12]: What is this value referencing?

**Comment by BertovDev** [comment]
Do we really need the Canvas Group component in the click GameObject?

File: client/Assets/Scripts/UI/ModelLink.cs (hunk 1 above)
Comment (AngieDutra) [reply client/Assets/Scripts/UI/ModelLink.cs:12]: The margin of difference allowed between the eventData pressed and released

**Comment by AngieDutra** [comment]
> Do we really need the Canvas Group component in the click GameObject?

Yes. The button has an image to be clickable. An image always goes attached with a CanvasGroup as a Unity rule
Captura de pantalla 2024-03-19 a la(s) 8 00 49 p  m

**Review by BertovDev — APPROVED** [review]
This is awesome!

**Review by agustinesco — APPROVED** [review]
LGTM! tested on the client and in an android build and the rotation worked great!

**Pull request merged**